resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesTaskSetManager.scala

```diff
@@ -52,12 +56,30 @@ private[spark] class KubernetesTaskSetManager(
           if (pendingTasksClusterNodeIP.nonEmpty) {
             logDebug(s"Got preferred task list $pendingTasksClusterNodeIP for executor host " +
               s"$executorIP using cluster node IP $clusterNodeIP")
+            pendingTasksClusterNodeIP
+          } else {
+            val clusterNodeFullName = inetAddressUtil.getFullHostName(clusterNodeIP)
```

Comment:
Yes, that is possible. In fact, the first commit of the original PR was written in that style. But, there was a [review comment](https://github.com/apache-spark-on-k8s/spark/pull/216/files/b3855d651a7f639a25546fd0ca526b0bf5dbdbf7#r113525080) suggesting that the current form is the preferred scala style. PTAL the prior review comment and let me know what you think.